props/blaster.h

```diff
@@ -260,9 +266,39 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
+  RefPtr<BufferedWavPlayer> tmp;
+  uint32_t len;
+
   void Loop() override {
     PropBase::Loop();
     PollNextAction();
+	  
+    STATE_MACHINE_BEGIN();
+    while (true) {
+      if (auto_firing_) {
+        tmp = GetWavPlayerPlaying(&SFX_auto);
+        // Set the length for WavLen<>
+        if (tmp) {
+          tmp->UpdateSaberBaseSoundInfo();
+        } else {
+          SaberBase::ClearSoundInfo();
+        }
+        if (millis() - auto_time_ > 1000 * tmp->length()) {
```

Comment:
A simpler way to do this might be to use tmp->pos().
If you keep track of the previous position, then you can just do:

float pos = tmp->pos();
if (pos < last_pos_) {
    // sound has wrapped around
}
last_pos_ = pos;

Reply:
trying this, but can't get   tmp = GetWavPlayerPlaying(&SFX_auto); unless it's inside the loop.
Can't pass 'pos' , running into a wall tonight.

Reply:
not sure what this comment means